postprocessing/pp-opus.c

```diff
@@ -19,48 +19,42 @@
 #include <string.h>
 #include <stdlib.h>
 
-#include <ogg/ogg.h>
-
+#include "pp-avformat.h"
 #include "pp-opus.h"
 #include "pp-opus-silence.h"
 #include "../debug.h"
 #include "../version.h"
 
+static AVFormatContext *fctx;
+static AVStream *vStream;
 
-/* OGG/Opus helpers */
-FILE *ogg_file = NULL;
-ogg_stream_state *stream = NULL;
+static const uint8_t opus_extradata[19] = {
+    'O', 'p', 'u', 's', 'H', 'e', 'a', 'd',
+    1, 2, 56, 1, 128, 187,
+    0, 0, 0, 0, 0,
+};
 
-void le32(unsigned char *p, int v);
-void le16(unsigned char *p, int v);
-ogg_packet *op_opushead(void);
-ogg_packet *op_opustags(char *metadata);
-ogg_packet *op_from_pkt(const unsigned char *pkt, int len);
-void op_free(ogg_packet *op);
-int ogg_write(void);
-int ogg_flush(void);
+int janus_pp_opus_create(char *destination, char *metadata) {
+	if(destination == NULL)
+		return -1;
 
+	/* WebM output */
+	fctx = janus_pp_create_avformatcontext("ogg", metadata, destination);
```

Comment:
I guess that if we wanted to optionally allow people to output the processing to `.mka`, as you said in #2658, we'd only need to make the "ogg" part a configurable string where we can put a different format target, correct? Same for videos, I guess, where VP8/VP9 could be outputted to `.mkv` rather than `.webm` as we do now.

Reply:
That's the plan, ideally should be easy to select the output format (mp4 or mkv or webm).

I'll try to make the postprocessor able to mux both audio and video in a single pass later probably. One step at time first though :)

Reply:
> I'll try to make the postprocessor able to mux both audio and video in a single pass later probably. One step at time first though :)

That's actually not needed, since our `.mjr` files only store one media stream anyway (audio, video or data). Any muxing after that should be done separately, e.g., using ffmpeg itself, and possibly using timing info from the `.mjr` header to sort out sync issues.